### PR TITLE
Binary Search Tree in C#

### PR DESCRIPTION
**Pull request by Omar-Hegazi**
Have you read the Contributing Guidelines on Pull Requests?
(Write your answer here.)
yes

Description
(Write your answer here.)
I added an implementation of Binary search tree in C#. I build a insert function that add new node to tree and delete function to remove node with specific value from tree. I built functions for inorder, preorder and postorder traversal. I add function to find min value, max value and find node with specific value.

Checklist
[yes] I've read the contribution guidelines.
[yes ] I've checked the issue list before deciding what to submit.
[ yes] I've edited the README.md and link to my code.
Related Issues or Pull Requests
(Write your answer here.)
#3678

### CODE REVIEW

**Comment by Omar-Hegazi** [comment]
@ankitaggarwal23  I added the issue. I didn't understand at first the meaning of the question. I thought it meant some sort of bug. I hope you review my PR.

File: C-Sharp/dp/BinarySearchTree.cs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+30
+60
+80
+70
+50
+Min Value:
+30
+Max Value:
+80
+*/
```
Comment (bhav09) [inline C-Sharp/dp/BinarySearchTree.cs:233]: Add a new line at EOF

File: C-Sharp/dp/BinarySearchTree.cs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            Console.WriteLine("Max Value:");
+            Console.WriteLine(BST.FindMax());
+
+
+
+        }
+    }
+}
+/*
+Sample Input: 50 30 20 4070 60 80
```
Comment (bhav09) [inline C-Sharp/dp/BinarySearchTree.cs:209]: Can you mention the complexity of the code?

**Review by bhav09 — CHANGES_REQUESTED** [review]
You can have a few more comments in your code. Reviewed.

**Review by ankitaggarwal23 — COMMENTED** [review]
Don't push or commit more than one code file in a single PR.  Open the new PR.

**Comment by ankitaggarwal23** [comment]
@

**Comment by Omar-Hegazi** [comment]
@ankitaggarwal23 So I'm supposed to wait two days for the review for the second time then you give me a review for not committing  more than one and i didn't  commit any file except the bst and modify the read me and this is the same code i committed before and you said that i only need to address the issue then you close my pr without any understanding. are we messing around here.

File: C-Sharp/dp/BinarySearchTree.cs (hunk 2 above)
Comment (Omar-Hegazi) [reply C-Sharp/dp/BinarySearchTree.cs:209]: I would but the pr already closed.

**Pull request closed without merging**